src/modules/pythonmonkey/pythonmonkey.cc

```diff
@@ -365,17 +399,30 @@ static PyObject *eval(PyObject *self, PyObject *args) {
       } /* filename */
     } /* fromPythonFrame */
   } /* eval options */
-    // initialize JS context
-  JS::SourceText<mozilla::Utf8Unit> source;
-  if (!source.init(GLOBAL_CX, code->getValue(), strlen(code->getValue()), JS::SourceOwnership::Borrowed)) {
+
+  // compile the code to execute
+  JS::RootedScript script(GLOBAL_CX);
+  JS::Rooted<JS::Value> *rval = new JS::Rooted<JS::Value>(GLOBAL_CX);
+  if (code) {
+    JS::SourceText<mozilla::Utf8Unit> source;
+    if (!source.init(GLOBAL_CX, code->getValue(), strlen(code->getValue()), JS::SourceOwnership::Borrowed)) {
+      setSpiderMonkeyException(GLOBAL_CX);
+      return NULL;
```

Comment:
```suggestion
      delete code;
      return NULL;
```

Reply:
nice catch! Thanks

Reply:
Changes made. Will they should up in this discussion? They do in gitlab. `54578b600c90af2c63fdcc9bd58eacac598c7a11`